lib/publiccloud/gce.pm

```diff
@@ -170,7 +171,8 @@ sub cleanup {
 
     my $region = $self->{provider_client}->{region};
     my $project = $self->{provider_client}->{project_id};
-    my $instance_id = $args->{my_instance}->{instance_id};
+    my $instance_id = ($args->{my_instance}->{instance_id} || $self->{my_instance}->{instance_id}[-1]);
```

Comment:
Hello Maurizio, I suggest you to do [what I did in ec2](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/115e3ac750c822eece081478443effe48e1a8593/lib/publiccloud/ec2.pm#L213). It simplifies the code quite a lot.

Reply:
Interesting and smart solution, I'd like more clarifications on it, but my idea is to preserve the `create_instances` possible **multiple instances** creation, for which I store each instance_id into that array, then `cleanup` pop out the last one [-1] that, when routine is triggered on-error, it is the guilty id where error occurred.